src/field/pokemon.ts

```diff
@@ -4172,7 +4221,7 @@ export abstract class Pokemon extends Phaser.GameObjects.Container {
    * @param reloadAssets Whether to reload the assets or not; defaults to `false`.
    */
   resetStatus(confusion: boolean = false, reloadAssets: boolean = false): void {
-    const lastStatus = this.status?.effect;
+    const lastStatus = this.getStatusEffect();
```

Comment:
```suggestion
    const lastStatus = this.getStatusEffect(true);
```
This should ignore comatose, right?
Which makes me wonder if comatose impacts the animation frameRate like normal sleep does. Probably need to double check that